nestjs/src/certificates/certificates.service.ts

```diff
@@ -82,4 +85,21 @@ export class CertificationsService {
       },
     };
   }
+
+  public async removeCertificate(studentId: number) {
+    const certificate = await this.certificateRepository.findOne({
+      where: { studentId },
+    });
+
+    if (!certificate) {
+      throw new NotFoundException('Certificate not found');
+    }
+
+    await Promise.all([
+      lastValueFrom(
+        this.httpService.delete(`${this.configService.awsServices.restApiUrl}/certificate/${certificate.s3Key}`),
+      ),
+      this.certificateRepository.remove(certificate),
+    ]);
```

Comment:
What's the point of using `lastValueFrom` here? Why can't it be just `Promise.all`?